Add top-level project dropdown

alkamin opened the pull request:
## Overview

This PR adds a dropdown to the nav-bar (in between 'Home' and 'Projects') that displays the currently active project (a concept we are introducing here) and options associated with that project. The dropdown also provides a quick way for a user to switch projects while not leaving the editor/browser.

The browse page now takes a route parameter, `projectid`, that will help handle this change. The current route structure is now `/browse/:projectid/:sceneid?`

To further facilitate this change, the `selectedScenesModal` modal has been modified reflect the single active project concept. After selecting scenes and clicking the button at the bottom of the scenes list, the user is presented with the list of scenes, and from there, they can add those scenes to the currently active project in a single click.

In addition, this PR also:
- does a bit of reorganization on the `browse.controller` (see notes)
- adds back the 'selectProjectModal' that was previously removed
- makes `/home` the default landing page

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

<img width="605" alt="screen shot 2017-03-22 at 1 07 03 pm" src="https://cloud.githubusercontent.com/assets/2442245/24210792/8ebb63b4-0f00-11e7-914c-dafec2f7c2ef.png">
<img width="546" alt="screen shot 2017-03-22 at 1 06 34 pm" src="https://cloud.githubusercontent.com/assets/2442245/24210796/93010bcc-0f00-11e7-9fcf-99a8b2ab2ee6.png">
<img width="539" alt="screen shot 2017-03-22 at 1 06 46 pm" src="https://cloud.githubusercontent.com/assets/2442245/24210797/9303b7d2-0f00-11e7-974c-e8fe7954c58d.png">
<img width="857" alt="screen shot 2017-03-22 at 1 07 26 pm" src="https://cloud.githubusercontent.com/assets/2442245/24210803/95824e1a-0f00-11e7-8402-13190cc628af.png">


### Notes

The browse page was creating a `queryParams` variable that contained all items from `$state.params`. This params list provided by angular also includes any parameters defined within the route. Because of this, route params were being persisted in the session storage and would overwrite parameters defined in the URL. For the case of the browse page, this was `projectid` and `sceneid`.

This PR separates the two concepts into `routeParams` and `queryParams` to prevent any conflation of the two sets of parameters. `routeParams` are never cached in session storage.

### More Notes

This PR doesn't address any other changes to the browse page, so there is no way to know what scenes are already in the current project when adding scenes.

## Testing Instructions

  * Log out and log in: ensure you are taken to the `Home` page
  * Click `View Projects` and go to the list of your projects and then select a project
  * Click the 'Editor' button, you should be taken to the browse page
    * Ensure that the project dropdown in the navbar shows the project you just selected
    * Ensure that that the project's UUID is in the route
    * Reload: ensure the project is still displayed
  * Click on a scene to view it's details, ensure the route is updated to reflect the selected scene
  * Reload: ensure that the scene details are still visible
  * Close the scene details pane, check that the route no longer has the scene id in it
  * Reload: ensure that the scene details are not visible
  * Select some scenes and click the button at the bottom
  * Check that the buttons on the bottom are now 'Cancel' and 'Add to <projectname>'
  * Check in a few ways that the scenes you have selected/unselected from the modal are successfully added to the project.
  * In the project dropdown, verify that the 'Add Scenes' option is disabled (if you're on the `/browse` page
  * Select 'Switch Project' and select a different project
  * Ensure that the route and dropdown title have changed to reflect the new current project
  * Try this a few more times with some scene selection/addition combinations to see that everything is working
  * Using the project dropdown, click 'Edit' and make sure that takes you to the editor
  * Check that 'Edit' is now disabled (since you're in the editor)
  * Check that 'Add Scenes` is now enabled and will take you back to the scene browser
  * Return to the editor and check that switching projects works as expected
  * If you have any more energy, do some other things

Connects #1260

notthatbreezy commented:
![should-go-browse](https://cloud.githubusercontent.com/assets/898060/24248904/aa7fd394-0fa7-11e7-8a74-abea0d5c2f84.png)

After hitting next here - this should go to the browse page for that project so you can add scenes, currently it just goes back to home

notthatbreezy commented:
Button I was talking about

![maybe-rename-button](https://cloud.githubusercontent.com/assets/898060/24249524/f5bafd50-0fa9-11e7-9d82-aeafe8eb2c60.png)

alkamin commented:
Changed the button text to 'View {{projectname}}' and fixed the redirect. Any attempt to go to browse without a project id set in the route will now redirect to the home page
